doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -144,61 +144,101 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
-   sort             : SProp | Prop | Set | Type
-   fix_bodies       : `fix_body`
-                    : `fix_body` with `fix_body` with … with `fix_body` for `ident`
-   cofix_bodies     : `cofix_body`
-                    : `cofix_body` with `cofix_body` with … with `cofix_body` for `ident`
-   fix_body         : `ident` `binders` [`annotation`] [: `term`] := `term`
-   cofix_body       : `ident` [`binders`] [: `term`] := `term`
-   annotation       : { struct `ident` }
-   match_item       : `term` [as `name`] [in `qualid` [`pattern` … `pattern`]]
-   dep_ret_type     : [as `name`] `return_type`
-   return_type      : return `term`
-   equation         : `mult_pattern` | … | `mult_pattern` => `term`
-   mult_pattern     : `pattern` , … , `pattern`
-   pattern          : `qualid` `pattern` … `pattern`
-                    : @ `qualid` `pattern` … `pattern`
-                    : `pattern` as `ident`
-                    : `pattern` % `ident`
-                    : `qualid`
-                    : _
-                    : `num`
-                    : ( `pattern` | … | `pattern` )
+   term : forall `open_binders` , `term`
+        : fun `open_binders` => `term`
+        : `term_let`
+        : if `term` `return_type_opt` then `term` else `term`
+        : `single_fix`
+        : `single_fix` with `fix_body_list` for `ident`
+        : `term100`
+   type_cstr : : `term`
+             : `empty`
+   name : _
+        : `ident`
+   return_type_opt : `as_name_opt` return `term100`
+                   : `empty`
+   as_name_opt : as `name`
+               : `empty`
+   binders_fixannot : { `ident` `impl_ident` } `binders_fixannot`
+                    : `fixannot`
+                    : `binder` `binders_fixannot`
+                    : `empty`
+   impl_ident : `empty`
+              : `name_list` : `term`
+              : `name_list`
+              : : `term`
+   fixannot : { struct `ident` }
+            : { wf `constr` `ident` }
+            : { measure `constr` `ident_opt` `constr_opt` }
+            : { struct `name` }
+            : `empty`
+   name_list : `name_list` `name`
+             : `name`
+   term100 : `term_cast`
+           : `term10` :>
+           : `term10`
+   term10 : `term9` `appl_arg_list`
+          : @ `qualid` `universe_annot_opt` `term9_list_opt`
+          : `term9`
+   ident_list : `ident_list` `ident`
+              : `ident`
+   appl_arg_list : `appl_arg_list` `appl_arg`
+                 : `appl_arg`
+   appl_arg : ( `ident` := `term` )
+            : `term9`
+   term9 : `term1`
+   term1 : `term0` .( `qualid` `appl_arg_list_opt` )
+         : `term0` .( @ `qualid` `term9_list_opt` )
+         : `term0` % `ident`
+         : `term0`
+   appl_arg_list_opt : `appl_arg_list_opt` `appl_arg`
+                     : `empty`
+   term9_list_opt : `term9_list_opt` `term9`
+                  : `empty`
+   empty :
+   term0 : `qualid` `universe_annot_opt`
+         : `sort`
+         : `numeral`
+         : `string`
```

Comment:
Notation related. Could be omitted from the main grammar and mentioned as an extension via `prodn` in the notation chapter. We may want to keep `numeral` here since it has a dedicated section below, and is initially bound to natural numbers, whereas `string` is not bound to anything initially.

Reply:
"bound to" meaning there is a definition of the grammar symbol in the doc?

How would you define `string`?  I don't see how I can make a single quote mark

```
string : " chars "
chars : chars char
char : <any character>
```

This doesn't include escaped characters, either.

Let me know what you want here.

Reply:
Not sure I understand your question. `string` is already defined in the manual, is it not? And my suggestion was rather to remove it from here. Numerals are indeed notations that represent natural numbers by default in Coq's prelude.